taichi/transforms/auto_diff.cpp

```diff
@@ -200,9 +200,18 @@ class ReplaceLocalVarWithStacks : public BasicStmtVisitor {
                          })
                          .empty();
     if (!load_only) {
-      alloc->replace_with(Stmt::make<StackAllocaStmt>(
-          alloc->ret_type.data_type,
-          get_current_program().config.ad_stack_size));
+      auto dtype = alloc->ret_type.data_type;
+      auto stack_alloca = Stmt::make<StackAllocaStmt>(
+          dtype, alloc->get_kernel()->program.config.ad_stack_size);
+      auto stack_alloca_ptr = stack_alloca.get();
+
+      alloc->replace_with(std::move(stack_alloca));
+
+      // Note that unlike AllocStmt, StackAllocaStmt does NOT have an 0 as
+      // initial value. Therefore here we push an initial 0 value.
```

Comment:
```suggestion
      // Note that unlike AllocaStmt, StackAllocaStmt does NOT have an 0 as
      // initial value. Therefore here we push an initial 0 value.
```
How about enforce `StackAllocaStmt` to be zero initialized over backends like CC did in #1752?

Reply:
There's nothing needed to be done in the backends. The instruction pushing a zero is emitted into the IR and `StackAllocaStmt` only needs to create an empty stack.